lib/stream/leo-stream.js

```diff
@@ -635,10 +637,14 @@ module.exports = function(configure) {
 					logger.time("kinesis request");
 					kinesis.putRecords({
 						Records: records.map((r) => {
+							let randomhashkey = 0
+							if(!opts.partitionHashKey || opts.partitionHashKey == "random") {
+								randomhashkey = uuidv4()	
+							}
 							return {
 								Data: r,
 								PartitionKey: "0",
-								ExplicitHashKey: (opts.partitionHashKey || 0).toString()
+								ExplicitHashKey: (opts.partitionHashKey || randomhashkey).toString()
```

Comment:
This line is an issue when allowing `"random"` for `opts.partitionHashKey`, it will set the `ExplicitHashKey` to be the value `random`